pkg/controller/history/controller_history.go

```diff
@@ -332,10 +333,14 @@ func (rh *realHistory) AdoptControllerRevision(parent metav1.Object, parentKind
 }
 
 func (rh *realHistory) ReleaseControllerRevision(parent metav1.Object, revision *apps.ControllerRevision) (*apps.ControllerRevision, error) {
+	dataBytes, err := patch.GenerateDeleteOwnerRefStrategicMergeBytes(parent.GetUID(), []types.UID{revision.UID})
```

Comment:
shouldn't the parent and revision UID be switched?

Reply:
yep! fixed it, but I will create a test related issue as well. 

https://github.com/kubernetes/kubernetes/issues/104787

I will see if I can fix it.